cpp/oneapi/dal/test/ccl_communicator.cpp

```diff
@@ -207,5 +231,33 @@ TEST_M(ccl_comm_test, "allgatherv") {
     }
 }
 
+TEST_M(ccl_comm_test, "send_receive_replace") {
+    auto comm = get_new_comm();
+    const std::int64_t count = 2;
+    const std::int64_t rank_count = comm.get_rank_count();
+    const std::int64_t rank = comm.get_rank();
+    const std::int64_t destination_rank = (rank == 0) ? (rank_count - 1) : (rank - 1);
+    const std::int64_t source_rank = (rank == (rank_count - 1)) ? 0 : (rank + 1);
+
+    std::vector<float> buffer(count);
```

Comment:
I would extend these tests with float_t and instantiate it for float and double in next PR

Reply:
Agree. Should be added for all tests but by a separate MR.